.github/workflows/e2e-pr.yml

```diff
@@ -0,0 +1,185 @@
+name: E2E
+
+on:
+  pull_request:
+
+env:
+  PERCY_POSTINSTALL_BROWSER: true
+  GH_PAGES_OWNER: blackbaud
+  GH_PAGES_REPO: skyux-pr-preview
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  install-deps:
+    name: Install and cache dependencies
+    runs-on: ubuntu-latest
+    outputs:
+      projects: ${{ steps.affected.outputs.projects }}
+      e2eMap: ${{ steps.affected.outputs.e2eMap }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+      - name: Rebase current branch
+        run: node ./scripts/rebase-pr.js
+      - name: Cache dependencies
+        id: cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            !node_modules/.cache/nx
+            /home/runner/.cache/Cypress
+          key: cache-pr-preview-${{ hashFiles('package-lock.json') }}
+      - name: npm install
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          set -x
+          node ./scripts/rebase-pr.js
+          npm ci
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        id: setSHAs
+        uses: nrwl/nx-set-shas@v2
+      - name: Check affected
+        id: affected
+        run: npx ts-node scripts/github-workflow-e2e-matrix
```

Comment:
Capturing the affected projects doesn't make much sense in the context of the "Install and cache dependencies" step. I'd move these outputs into a separate step designed to capture variables that will be shared across all steps.

Reply:
The name wasn't accurate for what this step does. I've updated the name.